### ISSUE TITLE
MethodLengthChecker interprets maxLength differently, depending on ignoreComments

### ISSUE THREAD

**Issue by Darhazer** [scalastyle/scalastyle#302]
I was working on adding the ignoreEmptyLines option (see #300). Unlike #301 I decided to do some refactoring to make the code more functional. The existing tests won't pass.
After some debugging, I found that the problem is actually in the existing code base, which have an off-by-one error.

See:
```scala
@Test def testOK(): Unit = {
  val source = """
package foobar

class F1() {
  def method1() = {
    def foobar() = { 5 }
    2
    3
    4
    5
    6
    7
  }
}
"""

  assertErrors(List(columnError(5, 6, List("7"))), source, Map("maxLength" -> "7", "ignoreComments" -> "false")) // PASS
}
```

same code, this time with ignoreComments:
```scala
@Test def testOK(): Unit = {
  val source = """
package foobar

class F1() {
  def method1() = {
    def foobar() = { 5 }
    2
    3
    4
    5
    6
    7
  }
}
"""

  assertErrors(List(columnError(5, 6, List("7"))), source, Map("maxLength" -> "7", "ignoreComments" -> "true")) // FAIL
  }
```

With ignoreComments, the check allows for max 8 lines
Without ignoreComments, the check allows for max 6 lines

For the record, [my version](https://github.com/scalastyle/scalastyle/compare/master...Darhazer:refactor-method-length-checker?expand=1) and a [more functional one](https://github.com/scalastyle/scalastyle/compare/master...Darhazer:immutable-method-length-checker?expand=1)

**Comment by canoztokmak** [comment]
Hey @Darhazer,

In PR #301 , I realized that issue and fixed it already. After that one is merged, this bug will disappear.

**Comment by Darhazer** [comment]
There's one more issue - since it uses `contains` to check if the string is in multiline comment, it skips lines that end with `/*`, although the line itself contains valid code

**Comment by canoztokmak** [comment]
I thought that was an implementation decision since the issue is already pointed out in the code as comment.

https://github.com/scalastyle/scalastyle/blob/b3c6c80e31d4d557c10c70ec19821c48bf3e05af/src/main/scala/org/scalastyle/scalariform/MethodLengthChecker.scala#L83

**Comment by canoztokmak** [comment]
can be closed since pr is merged..

**Comment by matthewfarwell** [comment]
Fixed by #301.
